File: samples/java8/src/test/java/com/alibaba/cloud/cms/samples/client/SimplePrinterTest.java
package com.alibaba.cloud.cms.samples.client;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * 简洁打印器测试
 * Simple printer tests
 */
class SimplePrinterTest {
    private SimplePrinter printer;
    private ObjectMapper objectMapper;

    @BeforeEach
    void setUp() {
        printer = new SimplePrinter();
        objectMapper = new ObjectMapper();
    }

    @Test
    void testProcessNullEvent() {
        String result = printer.processEvent(null);
        assertEquals("", result);
    }

    @Test
    void testProcessEventWithNullBody() {
        ChatEvent event = new ChatEvent();
        String result = printer.processEvent(event);
        assertEquals("", result);
    }

    @Test
    void testProcessEventWithSystemArtifacts() throws Exception {
        String json = "{\"messages\":[{\"role\":\"system\",\"artifacts\":[{\"parts\":[{\"kind\":\"text\",\"text\":\"Hello World\"}]}]}]}";
        JsonNode body = objectMapper.readTree(json);

        ChatEvent event = ChatEvent.fromResponse(body, json, 200);
        String result = printer.processEvent(event);

        assertEquals("Hello World", result);
        assertEquals("Hello World", printer.getFinalText());
    }

    @Test
    void testProcessEventIgnoresAssistantRole() throws Exception {
        String json = "{\"messages\":[{\"role\":\"assistant\",\"artifacts\":[{\"parts\":[{\"kind\":\"text\",\"text\":\"Should be ignored\"}]}]}]}";
        JsonNode body = objectMapper.readTree(json);

        ChatEvent event = ChatEvent.fromResponse(body, json, 200);
        String result = printer.processEvent(event);

        assertEquals("", result);
    }

    @Test
    void testDeduplication() throws Exception {
        String json = "{\"messages\":[{\"role\":\"system\",\"artifacts\":[{\"parts\":[{\"kind\":\"text\",\"text\":\"Duplicate\"}]}]}]}";
        JsonNode body = objectMapper.readTree(json);

        ChatEvent event = ChatEvent.fromResponse(body, json, 200);

        // Process same event twice
        String result1 = printer.processEvent(event);
        String result2 = printer.processEvent(event);

        assertEquals("Duplicate", result1);
        assertEquals("", result2); // Should be empty due to deduplication
        assertEquals("Duplicate", printer.getFinalText());
    }

    @Test
    void testReset() throws Exception {
        String json = "{\"messages\":[{\"role\":\"system\",\"artifacts\":[{\"parts\":[{\"kind\":\"text\",\"text\":\"Test\"}]}]}]}";
        JsonNode body = objectMapper.readTree(json);

        ChatEvent event = ChatEvent.fromResponse(body, json, 200);
        printer.processEvent(event);

        assertEquals("Test", printer.getFinalText());

        printer.reset();

        assertEquals("", printer.getFinalText());

        // After reset, same content should be processed again
        String result = printer.processEvent(event);
        assertEquals("Test", result);
    }

    @Test
    void testMultipleArtifacts() throws Exception {
        String json = "{\"messages\":[{\"role\":\"system\",\"artifacts\":[" +
                "{\"parts\":[{\"kind\":\"text\",\"text\":\"Part1\"}]}," +
                "{\"parts\":[{\"kind\":\"text\",\"text\":\"Part2\"}]}" +
                "]}]}";
        JsonNode body = objectMapper.readTree(json);

        ChatEvent event = ChatEvent.fromResponse(body, json, 200);
        String result = printer.processEvent(event);

        assertEquals("Part1Part2", result);
    }

    @Test
    void testIgnoresNonTextKind() throws Exception {
        String json = "{\"messages\":[{\"role\":\"system\",\"artifacts\":[{\"parts\":[{\"kind\":\"image\",\"url\":\"http://example.com\"}]}]}]}";
        JsonNode body = objectMapper.readTree(json);

        ChatEvent event = ChatEvent.fromResponse(body, json, 200);
        String result = printer.processEvent(event);

        assertEquals("", result);
    }
}

File: samples/typescript/tests/simple-printer.test.ts
/**
 * Tests for SimplePrinter
 */

import { describe, it, expect, beforeEach } from 'vitest';
import { SimplePrinter } from '../src/client/simple-printer.js';
import { ChatEvent } from '../src/client/agent-client.js';

describe('SimplePrinter', () => {
  let printer: SimplePrinter;

  beforeEach(() => {
    printer = new SimplePrinter();
  });

  it('should return empty string for null event', () => {
    const result = printer.processEvent(null);
    expect(result).toBe('');
  });

  it('should return empty string for event with null body', () => {
    const event: ChatEvent = {
      rawJson: '',
      statusCode: 200,
      isDone: false,
    };
    const result = printer.processEvent(event);
    expect(result).toBe('');
  });

  it('should extract text from system artifacts', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Hello World' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('Hello World');
    expect(printer.getFinalText()).toBe('Hello World');
  });

  it('should ignore assistant role', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'assistant',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Should be ignored' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('');
  });

  it('should deduplicate artifacts', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Duplicate' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result1 = printer.processEvent(event);
    const result2 = printer.processEvent(event);

    expect(result1).toBe('Duplicate');
    expect(result2).toBe(''); // Should be empty due to deduplication
    expect(printer.getFinalText()).toBe('Duplicate');
  });

  it('should reset buffer', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Test' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    printer.processEvent(event);
    expect(printer.getFinalText()).toBe('Test');

    printer.reset();
    expect(printer.getFinalText()).toBe('');

    // After reset, same content should be processed again
    const result = printer.processEvent(event);
    expect(result).toBe('Test');
  });

  it('should handle multiple artifacts', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Part1' }] },
              { parts: [{ kind: 'text', text: 'Part2' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('Part1Part2');
  });

  it('should return empty string for non-text/non-task_finished event type', () => {
    const event: ChatEvent = {
      event: 'interaction',
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Should be skipped' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('');
  });

  it('should process event with event field set to text', () => {
    const event: ChatEvent = {
      event: 'text',
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'Text event content' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('Text event content');
  });

  it('should process event with no event field (fallback)', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'text', text: 'No event field' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('No event field');
  });

  it('should ignore non-text kind', () => {
    const event: ChatEvent = {
      body: {
        messages: [
          {
            role: 'system',
            artifacts: [
              { parts: [{ kind: 'image', url: 'http://example.com' }] },
            ],
          },
        ],
      },
      rawJson: '{}',
      statusCode: 200,
      isDone: false,
    };

    const result = printer.processEvent(event);

    expect(result).toBe('');
  });
});
